pkg/controller/perconaservermongodb/mgo.go

```diff
@@ -238,22 +238,31 @@ func (r *ReconcilePerconaServerMongoDB) reconcileCluster(ctx context.Context, cr
 		err := cli.SetDefaultRWConcern(ctx, mongo.DefaultReadConcern, mongo.DefaultWriteConcern)
 		// SetDefaultRWConcern introduced in MongoDB 4.4
 		if err != nil && !strings.Contains(err.Error(), "CommandNotFound") {
-			return api.AppStateError, errors.Wrap(err, "set default RW concern")
+			return api.AppStateError, nil, errors.Wrap(err, "set default RW concern")
 		}
 	}
 
-	membersLive, err := r.updateConfigMembers(ctx, cli, cr, replset)
+	rsMembers, err := r.updateConfigMembers(ctx, cli, cr, replset)
 	if err != nil {
-		return api.AppStateError, errors.Wrap(err, "failed to update config members")
+		return api.AppStateError, nil, errors.Wrap(err, "failed to update config members")
+	}
+
+	membersLive := 0
+	for _, member := range rsMembers {
+		switch member.State {
+		case mongo.MemberStatePrimary, mongo.MemberStateSecondary, mongo.MemberStateArbiter:
```

Comment:
minor: Here we can also include the default case with a continue action.

Reply:
is there any practical difference?

Reply:
Only for better readability and clarity since unhandled cases are explicitly handled in a way (through the default). The behaviour remains the same!